app/src/db/migrations/20230518000000_008-filename-etag.js

```diff
@@ -0,0 +1,33 @@
+exports.up = function (knex) {
+  return Promise.resolve()
+    // add name column to object table
+    .then(() => knex.schema.alterTable('object', table => {
+      table.string('name', 1024);
+    }))
+    // backfill name from path column data
+    .then(() => knex('object').select(['id','path']))
+    .then((tuples) => {
+      const queries = tuples.map(tuple =>
+        knex('object')
+          .where('id', tuple.id)
+          .update({ 'name': tuple.path.match(/(?!.*\/)(.*)$/)[0] })
+      );
+      return Promise.all(queries);
```

Comment:
Optional: could in-line the tuples.map into the Promise.all(...) so that it doesn't need to be declared and saved into memory.